crates/wasm-wave/tests/ui.rs

```diff
@@ -0,0 +1,117 @@
+use anyhow::{Context, Result};
+use std::{collections::HashMap, fmt::Write, fs, path::Path, sync::OnceLock};
+
+use wasm_wave::{
+    parser::ParserError,
+    untyped::UntypedFuncCall,
+    value::{resolve_wit_func_type, FuncType, Value},
+    wasm::{DisplayValue, WasmFunc},
+};
+use wit_parser::Resolve;
+
+fn test(path: &Path) -> Result<String> {
+    let filename = path.file_name().unwrap().to_string_lossy();
+    let inputs = std::fs::read_to_string(path)?;
+    let mut output = String::new();
+    let out = &mut output;
+    let inputs = inputs.trim_end().trim_end_matches(';');
+    for mut input in inputs.split(";\n") {
```

Comment:
Looks like this needs to account for Windows `\r\n` as well.

Reply:
Done, but in a nasty way [b19582e](https://github.com/bytecodealliance/wasm-tools/pull/1606/commits/b19582e9b27fda697315a40d2a7ddc1ba1134228)

Reply:
Windows newlines ended up causing me too many headaches with these files, so I used `.gitattributes` so that they are not changed on windows: https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings